eclass/coreos-go-utils.eclass

```diff
@@ -87,7 +87,7 @@ go_export() {
 	# Remove certain flags from $LDFLAGS to fix validation errors in
 	# Go >= 1.15.5 like:
 	#   go build runtime/cgo: invalid flag in go:cgo_ldflag: -Wl,-O1
-	filter-ldflags "-Wl,-O1"
+	filter-ldflags "-Wl,-O*"
```

Comment:
Maybe something like this could work to filter all `-O?` away while still preserving any other parameters.
```suggestion
	export LDFLAGS="$(echo "$LDFLAGS" | sed 's/,-O.//g')"
	filter-ldflags "-Wl"
```

Reply:
Actually the Go compiler shouldn't reject `-Wl,-O1` but does so because of a typo https://github.com/golang/go/blob/master/src/cmd/go/internal/work/security.go#L182 (capital `L`). We should fix this upstream to be able to remove this filtering.

Reply:
Updated as your suggestion, but had to slightly change it like that:
```
export LDFLAGS="$(echo "$LDFLAGS" | sed 's/,-O[[:alnum:]]*//g')"
filter-ldflags "-Wl"
```
Will soon run another SDK build.

Reply:
Filed https://github.com/golang/go/pull/42631 but we can work around it for now

Reply:
SDK build passed